src/Control/Effect/Cut.hs

```diff
@@ -1,4 +1,14 @@
 {-# LANGUAGE DeriveFunctor, ExistentialQuantification, FlexibleContexts, GeneralizedNewtypeDeriving, StandaloneDeriving #-}
+
+{- | Provides an effect to delimit backtracking in a given nondeterministic context. This effect is used in concert with 'Control.Effect.Choose.Choose' or 'Control.Effect.NonDet.NonDet'.
```

Comment:
```suggestion
{- | Provides an effect to delimit backtracking in a given nondeterministic context. This effect is used in concert with 'Control.Effect.NonDet.NonDet'.
```

As with `Cull`, we only actually implement a carrier using `NonDet`. Further, it doesn’t actually make sense to combine `Cut` with only `Choose`, since its semantics involve failure.